### DISCUSSION TITLE
Build Error in Next.js 14 with nuqs 1.17.1 - 'cache' Not Exported from 'react'

### DISCUSSION THREAD

**Discussion by SergGrey1992** [47ng/nuqs#530] [Q&A] [answered]
Hello,

I am encountering an issue when trying to build my project using Next.js 14 with nuqs version 1.17.1.

During the build process with the `next build` command, I am facing the following error:

Failed to compile.

./node_modules/nuqs/dist/chunk-6GIZOTLC.js
Attempted import error: 'cache' is not exported from 'react' (imported as 'cache').

This error suggests that the nuqs package is trying to import `cache` from `react`, but there seems to be no such export in the React library.

![error](https://github.com/47ng/nuqs/assets/55402324/94998c61-6299-4956-bfb4-313e545cf70d)

![used](https://github.com/47ng/nuqs/assets/55402324/602d651a-948c-4d82-835d-0c5faffbea20)

![useddd](https://github.com/47ng/nuqs/assets/55402324/6bfba870-ab6f-42c8-affe-03fa71fd4ec9)

This occurs during the build phase and is preventing successful compilation. Here is a brief overview of my environment:

- Next.js version: 14.1.0
- Nuqs version: 1.17.1
- React version: 18.2.0
- Yarn version: 1.22.21

Could you please help me understand why this error is occurring and how to resolve it?

Thank you in advance for your assistance.

**Comment by franky47** [comment]
This usually works fine, Next.js embeds its own version of the React canary releases that includes this upcoming feature.

Could it be that yarn or your webpack config resolves React another way that doesn't point to that internal version?

**Reply by SergGrey1992** [reply]
I don't have enough experience to understand your answer((

I have a very simple webpack setup

![config](https://github.com/47ng/nuqs/assets/55402324/b3d9cc51-e04a-4087-9e28-949f91d3ffcc)

**Reply by SergGrey1992** [reply]
I tried upgrading to the following
"14.1.4" 
"14.2.0-canary.43"
same result

**Reply by franky47** [reply]
What happens if you don't use `nuqs` and instead try to import the `cache` function directly from react?

https://nextjs.org/docs/app/building-your-application/data-fetching/patterns#using-react-cache-server-only-and-the-preload-pattern

https://react.dev/reference/react/cache#cache

**Reply by SergGrey1992** [reply, marked as answer]
Yes, you were right! After commenting out all the code that used nuqs, the error disappeared.

However, new issues emerged, which I had to address. The main issues were:

Build optimization failed: found pages without a React Component as default export in various file paths (.src/pages/*). The solution was to recognize that the pages directory was being mistaken for routing pages. Changing pages to routes resolved this issue.

There was also a TypeScript error.

After addressing these issues, the build was successful!

Then, I decided to reintroduce the nuqs code, and to my surprise, the build continued to be successful!

Thank you so much for such a wonderful library!

**Reply by franky47** [reply]
Yeah I've had such cases where the compiler chokes on an error somewhere but reports something completely irrelevant, those bugs aren't nice. Glad you managed to figure it out!
